packages/aws-cdk-lib/aws-stepfunctions/lib/util.ts

```diff
@@ -1,4 +1,30 @@
+import { AwsOwnedEncryptionConfiguration } from './aokencryptionconfiguration';
+import { CustomerManagedEncryptionConfiguration } from './cmkencryptionconfiguration';
+import { CfnActivity, CfnStateMachine } from './stepfunctions.generated';
+
 export function noEmptyObject<A>(o: Record<string, A>): Record<string, A> | undefined {
   if (Object.keys(o).length === 0) { return undefined; }
   return o;
+}
+
+// eslint-disable-next-line max-len
+export function buildEncryptionConfiguration(encryptionConfiguration? : CustomerManagedEncryptionConfiguration | AwsOwnedEncryptionConfiguration) : CfnStateMachine.EncryptionConfigurationProperty | CfnActivity.EncryptionConfigurationProperty | undefined {
```

Comment:
Can the type here not just be`EncryptionConfiguration?  
```suggestion
export function buildEncryptionConfiguration(encryptionConfiguration? : EncryptionConfiguration) : CfnStateMachine.EncryptionConfigurationProperty | CfnActivity.EncryptionConfigurationProperty | undefined {
```

Also how come the return type can either be `CfnStateMachine.EncryptionConfigurationProperty | CfnActivity.EncryptionConfigurationProperty`?

Reply:
Please do not expose the Cfn interfaces in the L2 interface. Make a new interface, even if you feel it should match the L1.

Reply:
Updating function parameters to accept type of `EncryptionConfiguration `